### PR TITLE
M*LIB: non-inlined strings, FuriString primitive

### PR DESCRIPTION
**Pull request by DrZlo13**
# What's new

- FuriString primitive instead of M*LIB's string_t.

# Verification 

- Absolutely all code is affected, you need to check everything.

# Checklist (For Reviewer)

- [x] PR has description of feature/bug or link to Confluence/Jira task
- [x] Description contains actions to verify feature/bugfix
- [x] I've built this code, uploaded it to the device and verified feature/bugfix

### CODE REVIEW

File: scripts/fwsize.py [hunk 1]
```diff
@@ -38,13 +38,25 @@ def process_elf(self):
         return 0
 
     def process_bin(self):
+        BT_STACK_START = int("0x080D7000", 16)
+        BT_STACK_NAME = "stm32wb5x_BLE_Stack_light_fw"
```
Comment (hedger) [inline scripts/fwsize.py]: use variables from fbt environment? And pass them as args
COPRO_STACK_BIN, COPRO_STACK_ADDR, .bin parser

Comment (DrZlo13) [reply scripts/fwsize.py]: This feature will be made in another PR.

File: furi/core/string.c [hunk 2]
```diff
@@ -0,0 +1,302 @@
...
+
+static FuriStringUTF8State state_to_furi_state(m_str1ng_utf8_state_e state) {
+    switch(state) {
+    case M_STRING_UTF8_STARTING:
+        return FuriStringUTF8StateStarting;
+    case M_STRING_UTF8_DECODING_1:
+        return FuriStringUTF8StateDecoding1;
+    case M_STRING_UTF8_DECODING_2:
+        return FuriStringUTF8StateDecoding2;
+    case M_STRING_UTF8_DOCODING_3:
```
Comment (gsurkov) [inline furi/core/string.c:291]: Typo

Comment (skotopes) [reply furi/core/string.c:291]: Not in our code base.

**Review by skotopes — APPROVED** [review]

**Pull request merged**